### PR TITLE
Propagate secrets through spark config

### PR DESCRIPTION
**Pull request by tanya-borisova**
Propagate secrets for accessing Feature store and Data sources into the pipeline code through Spark config

### CODE REVIEW

File: infrastructure/transform/locals.tf [hunk 1]
```diff
@@ -65,4 +65,12 @@ locals {
     ]
   ]))
 
+  data_source_connection_secrets = flatten([for connection in var.data_source_connections : {
+    name     = connection.name
+    fqdn     = "{{secrets/${databricks_secret_scope.secrets.name}/flowehr-dbks-${connection.name}-fqdn}}"
```
Comment (t-young31) [inline infrastructure/transform/locals.tf:70]: could we have a comment about why they're in this absolutely nuts format?

**Review by t-young31 — APPROVED** [review]
LGTM!

File: infrastructure/transform/locals.tf (hunk 1 above)
Comment (tanya-borisova) [reply infrastructure/transform/locals.tf]: good point - I just moved the whole thing to where Spark config is, so hopefully it's more clear now

**Review by jongillham — APPROVED** [review]

**Review by t-young31 — APPROVED** [review]

**Pull request merged**